velox/functions/lib/aggregates/MinMaxAggregateBase.cpp

```diff
@@ -0,0 +1,649 @@
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    addRawInput(groups, rows, args, mayPushdown);
+  }
+
+  void addSingleGroupRawInput(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdateSingleGroup<std::function<bool(int32_t)>, nullHandlingMode>(
+        group, rows, args[0], [](int32_t compareResult) {
+          return compareResult < 0;
+        });
+  }
+
+  void addSingleGroupIntermediateResults(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    addSingleGroupRawInput(group, rows, args, mayPushdown);
+  }
+};
+
+template <CompareFlags::NullHandlingMode nullHandlingMode>
+class NonNumericMinAggregate : public NonNumericMinMaxAggregateBase {
+ public:
+  explicit NonNumericMinAggregate(
+      const TypePtr& resultType,
+      bool throwOnNestedNulls)
+      : NonNumericMinMaxAggregateBase(resultType, throwOnNestedNulls) {}
+
+  void addRawInput(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdate<std::function<bool(int32_t)>, nullHandlingMode>(
+        groups, rows, args[0], [](int32_t compareResult) {
+          return compareResult > 0;
+        });
+  }
+
+  void addIntermediateResults(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    addRawInput(groups, rows, args, mayPushdown);
+  }
+
+  void addSingleGroupRawInput(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdateSingleGroup<std::function<bool(int32_t)>, nullHandlingMode>(
+        group, rows, args[0], [](int32_t compareResult) {
+          return compareResult > 0;
+        });
+  }
+
+  void addSingleGroupIntermediateResults(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    addSingleGroupRawInput(group, rows, args, mayPushdown);
+  }
+};
+
+template <
+    template <typename T>
+    class TNumeric,
+    template <CompareFlags::NullHandlingMode nullHandlingMode>
+    typename TNonNumeric>
+exec::AggregateFunctionFactory getMinMaxFunctionFactoryInternal(
+    const std::string& name,
+    CompareFlags::NullHandlingMode nullHandlingMode,
+    TimestampPrecision precision) {
+  auto factory = [name, nullHandlingMode, precision](
+                     core::AggregationNode::Step step,
+                     std::vector<TypePtr> argTypes,
+                     const TypePtr& resultType,
+                     const core::QueryConfig& /*config*/)
+      -> std::unique_ptr<exec::Aggregate> {
+    const bool throwOnNestedNulls =
```

Comment:
This doesn't looks right. This could be a pre-existing bug. throwOnNestedNulls is not the same as NullHandlingMode.

Reply:
I updated to make `throwOnNestedNulls = false` when `NullHandlingMode == kNullAsValue`.

Reply:
BTW, as you mentioned, the behavior here does not align with Presto's.
```
presto> select min(x), max(x) from (values (array[1, 2]), (array[6, 7]), (array[3, null])) as t(x);
 _col0  | _col1
--------+--------
 [1, 2] | [6, 7]
(1 row)
```
Velox would throw exceptions for above query. I think it's related to the check `checkNestedNulls`.